examples/xyz-retina.js

```diff
@@ -20,6 +21,8 @@ const map = new Map({
         url:
           'https://api.maptiler.com/maps/outdoor-v2/256/{z}/{x}/{y}@2x.png?key=' +
           key,
+        tileSize: 512,
+        tileGrid: createXYZ({tileSize: 256, maxZoom: 22}),
```

Comment:
Why is this needed?

Reply:
It is equivalent to specifying `tilePixelRatio: 2` in `ol/source/XYZ`.  It is not needed in that example any more than `tilePixelRatio` was when the example used `ol/source/XYZ` as there is no reprojection, but as then it is the correct way to specify the source in case it is reprojected.

Since DataTile sources can have different pixel ratios in each dimension there is no `tilePixelRatio` option, see #13648.  However, I do think having a `tilePixelRatio` option for `ol/source/ImageTile` would be a nice future enhancement, where unless both `tileSize` and `tileGrid` were specified they could be calculated and passed to `super()`.

Reply:
I have now added a `tilePixelRatio` option so the source can be set up in a simpler way consistent with `ol/source/XYZ` (based on the code from #16279 which ensures data sizes are integer even if tile grid sizes are not).  The new hidpi map rendering test is using it.